src/sherpa_ai/utils.py

```diff
@@ -325,3 +329,124 @@ def check_if_number_exist(result: str, source: str):
+    #         """
+    nlp = spacy.load("en_core_web_sm")
+    doc = nlp(text)
+    entity_types = ["NORP", "ORG", "GPE", "LAW"]
+    filtered_entities = [ent.text for ent in doc.ents if ent.label_ in entity_types]
+
+    print("############# Entities gathered ##########################")
+    print(text)
+    print(filtered_entities)
+    print("#######################################")
+
+    return filtered_entities
+
+
+def is_subset(str1, str2):
+    set1 = set(str1)
+    set2 = set(str2)
+    return set1.issubset(set2)
+
+
+def text_similarity_by_llm(check_entity: [str], source_entity: [str]):
+    return {"entity_exisist": True, "messages": ""}
+
+
+def text_similarity_by_metrics(check_entity: [str], source_entity: [str]):
+    error_entity = []
+    message = ""
+    levenshtein_constant = 0.5
+
+    print("********************************************")
+    print("********************************************")
+    print(source_entity)
+    print(check_entity)
+    print("********************************************")
+    print("********************************************")
+    for entity in source_entity:
+        metrics_value = 0
+        for sub_entitiy in check_entity:
+            word1 = entity
+            word2 = sub_entitiy
+
+            combined_similarity = string_comparison_with_jaccard_and_levenshtein(
+                word1=word1, word2=word2, levenshtein_constant=levenshtein_constant
+            )
+
+            print("========================================================")
+            if metrics_value < combined_similarity:
+                metrics_value = combined_similarity
+        if metrics_value < 0.75:
+            return error_entity.append(entity)
+
+    if len(error_entity) > 0:
+        for entity in error_entity:
+            message += entity + ", "
+        message = f"remember to address these entities {message} in final the answer."
+        return {"entity_exisist": False, "messages": message}
+    return {"entity_exisist": True, "messages": message}
+
+
+def text_similarity(check_entity: [str], source_entity: [str]):
+    error_entity = []
+    message = ""
+    for entity in source_entity:
+        print(entity)
+        if entity not in check_entity:
+            error_entity.append(entity)
+    if len(error_entity) > 0:
+        for entity in error_entity:
+            message += entity + ", "
+        message = f"remember to address these entities {message} in final the answer."
+        return {"entity_exisist": False, "messages": message}
+    return {"entity_exisist": True, "messages": message}
+
+
+def check_entities_match(result: str, source: str, stage: int):
```

Comment:
I suggest we remove this function unless we have a clear use for it. Right now there's only a test calling it.

If we need it, 
- let's discuss
- The "stage" part is hard to understand and would have to be replaced by meaningful named values